src/Application/Commands/DeleteShoppingItem/DeleteShoppingItemCommand.php

```diff
@@ -33,6 +45,9 @@ public function execute(DeleteShoppingItemModel $model): void
             new ShoppingItemSelector($model->getItem())
         );
 
+        if ($item === null) {
```

Comment:
Required: this `if` statement should not be required. As you've already run the validation above (line 38), by this point in the execution nothing should go wrong, because the validator should have rejected all the unhappy paths. The fact you've had to put this `if` statement in is an indicator that the validator implementation isn't fully working (see a later comment).

Reply:
Makes sense now looking back at it!